src/query/execute_statement.c

```diff
@@ -181,6 +182,15 @@ static void init_compile_context (PARSER_CONTEXT * parser);
 
 static int do_select_internal (PARSER_CONTEXT * parser, PT_NODE * statement, bool for_ins_upd);
 
+static int get_dblink_password_encrypt (PARSER_CONTEXT * parser, const char *passwd, DB_VALUE * encrypt_val,
+					bool is_parser);
+static int get_dblink_password_decrypt (PARSER_CONTEXT * parser, const char *passwd_cipher, DB_VALUE * decrypt_val);
+static int shake_4_encrypt (const char *passwd, char *confused, struct timeval *chk_time);
+static int reverse_shake_4_decrypt (char *confused, char *passwd);
```

Comment:
I don't think these functions are placed in the proper file. The parser context is only used for throwing errors and there seems not related to executing statements.

Reply:
The below two functions have been moved to crypt_opfunc.c.
  shake_4_encrypt(), reverse_shake_4_decrypt()
  
The two functions below remove the parser parameter and add an error code.
  get_dblink_password_encrypt (), get_dblink_password_decrypt ()